Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3708,6 +3697,17 @@ void MediaPlayerPrivateGStreamer::setVideoSourceOrientation(ImageOrientation ori
 #if USE(TEXTURE_MAPPER_GL)
     updateTextureMapperFlags();
 #endif
+
+    // If the video is tagged as rotated 90 or 270 degrees, swap width and height.
+    if (m_videoSourceOrientation.usesWidthAsHeight())
+        m_videoSize = m_videoSize.transposedSize();
+
+    GST_DEBUG_OBJECT(pipeline(), "Enqueuing and waiting for main-thread task to call sizeChanged()...");
+    bool sizeChangedProcessed = m_sinkTaskQueue.enqueueTaskAndWait<AbortableTaskQueue::Void>([this] {
+        m_player->sizeChanged();
+        return AbortableTaskQueue::Void();
+    }).has_value();
+    GST_DEBUG_OBJECT(pipeline(), "Finished waiting for main-thread task to call sizeChanged()... %s", sizeChangedProcessed ? "sizeChanged() was called." : "task queue aborted by flush");
```

Comment:
setVideoSourceOrientation() is also called by updateVideoSizeAndOrientationFromCaps() who is called from a main-thread task in a lambda enqueued in videoSinkCapsChanged(). In that case this code would deadlock, as the main thread would be waiting for itself.

Reply:
So if we're in main thread we shouldn't call `m_sinkTaskQueue.enqueueTaskAndWait`?

Reply:
Also with this patch there's potentially 2 `sizeChanged` notifications in the `updateVideoSizeAndOrientationFromCaps()` code path... I'll rework this....